plugins/modules/event.py

```diff
@@ -0,0 +1,296 @@
+    check: awesome_check
+    check_attributes:
+      duration: 1.945
+      executed: 1522100915
+      history:
+        - executed: 1552505193
+          status: 1
+      issued: 1552506034
+      last_ok: 1552506033
+      output: '10'
+      state: 'passing'
+      status: 'ok'
+      total_state_change: 0
+    metric_attributes:
+      handlers:
+        - handler1
+        - handler2
+      points:
+        - name: "sensu-go-sandbox.curl_timings.time_total"
+          tags:
+            - name: "response_time_in_ms"
+              value: 101
+          timestamp: 1552506033
+          value: 0.005
+        - name: "sensu-go-sandbox.curl_timings.time_namelookup"
+          tags:
+            - name: "namelookup_time_in_ms"
+              value: 57
+          timestamp: 1552506033
+          value: 0.004
+'''
+
+RETURN = '''
+object:
+  description: object representing Sensu event
+  returned: always
+  type: dict
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+
+from ansible_collections.sensu.sensu_go.plugins.module_utils import (
+    arguments, errors, utils
+)
+
+STATUS_MAP = {
+    'ok': 0,
+    'warning': 1,
+    'critical': 2,
+    'unknown': 3,
+}
+
+
+def _get_check(client, check):
+    check_path = '/checks/{0}'.format(check)
+    resp = client.get(check_path)
+    if resp.status != 200:
+        raise errors.SyncError("Check with name '{0}' does not exist on remote.".format(check))
+    return resp.json
+
+
+def _get_entity(client, entity):
+    entity_path = '/entities/{0}'.format(entity)
+    resp = client.get(entity_path)
+    if resp.status != 200:
+        raise errors.SyncError("Entity with name '{0}' does not exist on remote.".format(entity))
+    return resp.json
+
+
+def _update_payload_with_metric_attributes(payload, metric_attributes):
+    if metric_attributes:
+        payload['metrics'] = arguments.get_spec_payload(metric_attributes,
+                                                        *metric_attributes.keys())
+    return payload
+
+
+def _update_payload_with_check_attributes(payload, check_attributes):
+    if check_attributes:
+        if check_attributes['status']:
+            check_attributes['status'] = STATUS_MAP[check_attributes['status']]
+
+        filtered_attributes = arguments.get_spec_payload(check_attributes, *check_attributes.keys())
+        payload['check'].update(filtered_attributes)
+    return payload
+
+
+def _build_api_payload(client, params):
+    payload = arguments.get_spec_payload(params, 'timestamp')
+    payload['metadata'] = dict(
+        namespace=params['auth']['namespace']
+    )
+    payload['entity'] = _get_entity(client, params['entity'])
+    payload['check'] = _get_check(client, params['check'])
+
+    payload = _update_payload_with_check_attributes(payload, params['check_attributes'])
+    payload = _update_payload_with_metric_attributes(payload, params['metric_attributes'])
+    return payload
+
+
+def sync(client, path, payload, check_mode):
```

Comment:
The name of this function is a bit misleading. We are not actually syncing anything, we simply send the event. This is why I would suggest renaming this into `send` or something similar.